app/models/set_member_subject.rb

```diff
@@ -15,6 +16,7 @@ class SetMemberSubject < ActiveRecord::Base
   can_through_parent :subject_set, :update, :show, :destroy, :index, :update_links,
     :destroy_links
 
+  before_save :timestamp_newly_retired_workflows
```

Comment:
Yes. Personally, I'm in favour of an explicit method. I think I let myself damage the design in favour of not changing too much specs (laziness, basically). That's because it bites with factory_girl (can't do `create(:set_member_subject, retired_workflows: [workflow])` anymore), as well as the JSONAPI controller magic.

Do you know how to make [the SMS controller](https://github.com/zooniverse/Panoptes/blob/master/app/controllers/api/v1/set_member_subjects_controller.rb#L6-L7) use that method instead of the association to manage the links?